File: server-scala-play-server/app/api/OpenAIApiController.scala
package api

import org.openapitools.OpenApiExceptions
import javax.inject.{Inject, Singleton}
import play.api.libs.json._
import play.api.mvc._
import model.WrappedProxyItemResponse

@javax.annotation.Generated(value = Array("org.openapitools.codegen.languages.ScalaPlayFrameworkServerCodegen"), date = "2026-02-11T19:31:43.310890579Z[Etc/UTC]", comments = "Generator version: 7.20.0-SNAPSHOT")
@Singleton
class OpenAIApiController @Inject()(cc: ControllerComponents, api: OpenAIApi) extends AbstractController(cc) {
  /**
    * POST /api/3.18/openai/v1/images/generations?accountId=[value]&postBody=[value]&returnRawResponse=[value]
    */
  def imageGeneration(): Action[AnyContent] = Action { request =>
    def executeApi(): WrappedProxyItemResponse = {
      val accountId = request.getQueryString("accountId")
        .map(value => value.toLong)
        .getOrElse {
          throw new OpenApiExceptions.MissingRequiredParameterException("accountId", "query string")
        }
        
      val postBody = request.getQueryString("postBody")
        .getOrElse {
          throw new OpenApiExceptions.MissingRequiredParameterException("postBody", "query string")
        }
        
      val returnRawResponse = request.getQueryString("returnRawResponse")
        .map(value => value.toBoolean)
        
      api.imageGeneration(accountId, postBody, returnRawResponse)
    }

    val result = executeApi()
    val json = Json.toJson(result)
    Ok(json)
  }

  private def splitCollectionParam(paramValues: String, collectionFormat: String): List[String] = {
    val splitBy =
      collectionFormat match {
        case "csv" => ",+"
        case "tsv" => "\t+"
        case "ssv" => " +"
        case "pipes" => "|+"
      }

    paramValues.split(splitBy).toList
  }
}

File: server-java-play-framework/app/controllers/OpenAiApiController.java
package controllers;

import java.math.BigDecimal;
import apimodels.WrappedProxyItemResponse;

import com.typesafe.config.Config;
import play.mvc.Controller;
import play.mvc.Result;
import play.mvc.Http;
import java.util.List;
import java.util.Map;
import java.util.ArrayList;
import java.util.LinkedHashSet;
import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.databind.JsonNode;
import com.google.inject.Inject;
import java.io.File;
import play.libs.Files.TemporaryFile;
import openapitools.OpenAPIUtils;
import com.fasterxml.jackson.core.type.TypeReference;

import javax.validation.constraints.*;
import javax.validation.Valid;
import com.typesafe.config.Config;

import openapitools.OpenAPIUtils.ApiAction;

@javax.annotation.Generated(value = "org.openapitools.codegen.languages.JavaPlayFrameworkCodegen", date = "2026-02-09T20:49:20.380387445Z[Etc/UTC]", comments = "Generator version: 7.20.0-SNAPSHOT")
public class OpenAiApiController extends Controller {
    private final OpenAiApiControllerImpInterface imp;
    private final ObjectMapper mapper;
    private final Config configuration;

    @Inject
    private OpenAiApiController(Config configuration, OpenAiApiControllerImpInterface imp) {
        this.imp = imp;
        mapper = new ObjectMapper();
        this.configuration = configuration;
    }

    @ApiAction
    public Result imageGeneration(Http.Request request, BigDecimal version) throws Exception {
        String valueaccountId = request.getQueryString("accountId");
        Long accountId;
        if (valueaccountId != null) {
            accountId = Long.parseLong(valueaccountId);
        } else {
            throw new IllegalArgumentException("'accountId' parameter is required");
        }
        String valuepostBody = request.getQueryString("postBody");
        String postBody;
        if (valuepostBody != null) {
            postBody = valuepostBody;
        } else {
            throw new IllegalArgumentException("'postBody' parameter is required");
        }
        String valuereturnRawResponse = request.getQueryString("returnRawResponse");
        Boolean returnRawResponse;
        if (valuereturnRawResponse != null) {
            returnRawResponse = Boolean.valueOf(valuereturnRawResponse);
        } else {
            returnRawResponse = null;
        }
        return imp.imageGenerationHttp(request, version, accountId, postBody, returnRawResponse);
    }

}
